ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -0,0 +1,132 @@
+        Ensure.argIsNotNull(fileSourceInfo, "fileSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        fileSourceInfo.validate();
+        ingestionProperties.validate();
+        try {
+            StreamSourceInfo streamSourceInfo = IngestionUtils.fileToStream(fileSourceInfo);
+            return ingestFromStream(streamSourceInfo, ingestionProperties);
+        } catch (FileNotFoundException e) {
+            log.error("File not found when ingesting a file.", e);
+            throw new IngestionClientException("IO exception - check file path.", e);
+        }
+
+    }
+
+    @Override
+    public IngestionResult ingestFromBlob(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(blobSourceInfo, "blobSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        blobSourceInfo.validate();
+        ingestionProperties.validate();
+
+        //if it's a blob we ingest using the queued client
+        return queuedIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+    }
+
+    @Override
+    public IngestionResult ingestFromResultSet(ResultSetSourceInfo resultSetSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(resultSetSourceInfo, "resultSetSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        resultSetSourceInfo.validate();
+        ingestionProperties.validate();
+        try {
+            StreamSourceInfo streamSourceInfo = IngestionUtils.resultSetToStream(resultSetSourceInfo);
+            return ingestFromStream(streamSourceInfo, ingestionProperties);
+        } catch (IOException ex) {
+            String msg = "Failed to read from ResultSet.";
+            log.error(msg, ex);
+            throw new IngestionClientException(msg, ex);
+        }
+    }
+
+    @Override
+    public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(streamSourceInfo, "streamSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        streamSourceInfo.validate();
+        ingestionProperties.validate();
+
+        if (streamSourceInfo.isLeaveOpen()) {
+            throw new UnsupportedOperationException("Stream can't be Leave Open in ManagedStreamingIngestClient");
+        }
+
+        for (int i = 0; i < MAX_RETRY_CALLS; i++) {
+            try {
+                return streamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties);
+            } catch (Exception e) {
```

Comment:
Are there any exceptions that will cause us to go straight to queued?

Reply:
In C# we have the method IsPermanent (which just calls not on IsTransient) and seems to have some heuristics about what exceptions are permanent or not.
In case they are permanent it doesn't go to the queue, it just rethrows it.
None of the heuristics in the dotnet file seemed relevant here so I decided to skip it, but if you have some idea when we might want to go to the queue instantly or bail entirely I'd love to hear.

Reply:
I don't familiar with all the streaming ingestion errors but I think there are some errors that worth to check... you should consult about that with @ArielYehezkely  and\or Valentina from the Engine team

Reply:
I'm not sure if thats the case in streamingIngestion but for commands and querying - since version 2.3.0 of the SDK the server response returns errors as JSON with isPermanent field in them, pull latest and check if it has it here too or ask Valentina for that as well

Reply:
Ok I added support for the JSON permanent field, anything else?